internal/config/main.go

```diff
@@ -0,0 +1,25 @@
+package config
+
+import (
+	"go.uber.org/zap"
+	"time"
+)
+
+type StorageConfigInterface interface {
+	GetLogger() *zap.SugaredLogger
+
+	GetVaultBaseUrl() string
+	GetToken() string
+
+	GetApproleLoginPath() string
+	GetApproleLogoutPath() string
+	GetApproleRoleId() string
+	GetApproleSecretId() string
+
+	GetSecretsPath() string
+	GetPathPrefix() string
+	GetInsecureSkipVerify() bool
+
+	GetLockTimeout() time.Duration
```

Comment:
I had to fully qualify the `time.Duration` type name here, because `StorageConfigInterface` is moved into `package config` (with an alias from the top level `package certmagic_vault_storage`). The `Duration` type is defined in `package certmagic_vault_storage`.

The reason I put StorageConfigInterface into a separate package was so that the vault client (internal/client/main.go) could depend on it without introducing a circular dependency:

`certmagic_vault_interface` -> `internal` -> `config`

Should I move `type Duration time.Duration` into the config package?

Reply:
To keep it simple, my recommendation would be to move `StorageConfigInterface` back to `types.go`, change `StorageConfigInterface.GetLockTimeout()` & `StorageConfigInterface.GetLockPollingInterval()` to return `Duration` types (and comment out the duplicate definition in `main.go` for `StorageConfigInterface`).

That way, there's no (or very minimal) additional changes required for the `caddy-vault-storage` plugin.

Reply:
@kmott FYI- the caddy-vault-storage plugin compiles without any changes.

Reply:
@sundbry I get this error when I try to build locally, how did you modify `caddy-vault-storage` plugin?

```
$ make build
INFO [config_reader] Config search paths: [./ /home/kmott/GolandProjects/mywordpress-io/caddy-vault-storage /home/kmott/GolandProjects/mywordpress-io /home/kmott/GolandProjects /home/kmott /home /] 
INFO [lintersdb] Active 6 linters: [errcheck gosimple govet ineffassign staticcheck unused] 
INFO [loader] Go packages loading at mode 575 (exports_file|imports|name|types_sizes|compiled_files|deps|files) took 291.014892ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 786.922µs 
INFO [linters_context/goanalysis] analyzers took 5.412088ms with top 10 stages: buildir: 577.138µs, fact_deprecated: 259.127µs, inspect: 255.192µs, SA4008: 248.429µs, tokenfileanalyzer: 129.916µs, fact_purity: 123.752µs, printf: 110.667µs, typedness: 105.456µs, nilness: 103.743µs, SA5012: 102.848µs 
INFO [runner] Issues before processing: 570, after processing: 4 
INFO [runner] Processors filtering stat (out/in): skip_dirs: 570/570, identifier_marker: 570/570, exclude: 570/570, max_from_linter: 4/4, severity-rules: 4/4, path_prettifier: 570/570, autogenerated_exclude: 570/570, exclude-rules: 570/570, uniq_by_line: 4/570, sort_results: 4/4, fixer: 4/4, cgo: 570/570, filename_unadjuster: 570/570, diff: 4/4, max_same_issues: 4/4, path_shortener: 4/4, skip_files: 570/570, nolint: 570/570, max_per_file_from_linter: 4/4, source_code: 4/4, path_prefixer: 4/4 
INFO [runner] processing took 162.800347ms with stages: exclude-rules: 123.976747ms, identifier_marker: 35.872715ms, nolint: 2.211057ms, path_prettifier: 385.239µs, source_code: 125.988µs, uniq_by_line: 85.778µs, skip_dirs: 47.172µs, autogenerated_exclude: 31.918µs, cgo: 28.631µs, filename_unadjuster: 19.62µs, max_same_issues: 5.894µs, max_per_file_from_linter: 2.632µs, path_shortener: 2.364µs, max_from_linter: 1.717µs, exclude: 566ns, sort_results: 525ns, fixer: 465ns, diff: 442ns, severity-rules: 343ns, skip_files: 304ns, path_prefixer: 230ns 
INFO [runner] linters took 510.10557ms with stages: goanalysis_metalinter: 347.135864ms 
main.go:1: : # github.com/mywordpress-io/caddy-vault-storage
./main.go:76:58: cannot use s (variable of type *Storage) as config.StorageConfigInterface value in argument to certmagic_vault_storage.NewStorage: *Storage does not implement config.StorageConfigInterface (wrong type for method GetLockPollingInterval)
                have GetLockPollingInterval() certmagic_vault_storage.Duration
                want GetLockPollingInterval() time.Duration (typecheck)
package caddy_vault_storage
main_suite_test.go:7:2: could not import github.com/mywordpress-io/caddy-vault-storage (-: # github.com/mywordpress-io/caddy-vault-storage
./main.go:76:58: cannot use s (variable of type *Storage) as config.StorageConfigInterface value in argument to certmagic_vault_storage.NewStorage: *Storage does not implement config.StorageConfigInterface (wrong type for method GetLockPollingInterval)
                have GetLockPollingInterval() certmagic_vault_storage.Duration
                want GetLockPollingInterval() time.Duration) (typecheck)
        "github.com/mywordpress-io/caddy-vault-storage"
        ^
main_suite_test.go:49:24: undefined: caddy_vault_storage (typecheck)
        caddyVaultStorage := &caddy_vault_storage.Storage{
                              ^
main_test.go:189:25: undefined: caddy_vault_storage (typecheck)
                caddyVaultStorage := &caddy_vault_storage.Storage{
                                      ^
INFO File cache stats: 3 entries of total size 17.2KiB 
INFO Memory: 10 samples, avg is 59.6MB, max is 94.0MB 
INFO Execution took 809.316134ms                  
make: *** [Makefile:11: lint] Error 1
```

Reply:
I just tested it directly from my main caddy build, via this commit / go-1.20, which did not emit any warnings: https://github.com/caddyserver/caddy/commit/7a5c5444f8681a42db24a62e849b4a051237a389

I can fix this up without moving the StorageConfigInterface into a separate package.

Reply:
Sounds great, let me know once you have more changes to review.  Thank you!